google-cloud-storage/lib/google/cloud/storage/service.rb

```diff
@@ -421,6 +429,10 @@ def rewrite_file source_bucket_name,
                          token: nil,
                          user_project: nil
           key_options = rewrite_key_options source_key, destination_key
+          
+          is_idempotent = Retry.retry?(if_generation_match: if_generation_match)
```

Comment:
Note that this should be for the destination file, not the source-- not sure how this is distinguished in Ruby.